servers/src/main/java/tachyon/worker/block/BlockStoreEventListener.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block;
+
+/**
+ * A listener interface for receiving meta data mutation events of {@link BlockStore}.
+ * <p>
+ * All methods may be called concurrently, thus listener implementation needs to ensure
+ * thread-safety.
+ */
+public interface BlockStoreEventListener {
+
+  /**
+   * Actions when accessing a block.
+   *
+   * @param userId the ID of the user to access this block
+   * @param blockId the ID of the block to access
+   */
+  void onAccessBlock(long userId, long blockId);
+
+  /**
+   * Actions when committing a temporary block to a {@link BlockStoreLocation}.
```

Comment:
Maybe specify that the callback occurs after the actual event is completed, and it is guaranteed the operation succeeded.